### PR TITLE
Interface for Optional Rules, add 18 Los Angeles optional company LA Title

### LINKED ISSUE #1082: Feature suggestion: It should be possible to select variants of 18xx when creating a game
Some 18xx has variants/optional rules that is of interest to try. 
Example:
- 18AL: Use "hard-rust" instead of "obsoleteness" when buying 7T
- 18MEX: Allow corporations to lay 3 yellow tiles instead of 2 during its first OR

It would be nice if when creating the game it would be possible to select which optional rules (if any) that should be used for the game. The selection could be shown in the description for the game.
It should be possible to select multiple optional rules (each game should check if a combination is invalid).

I see no use case for changing optional rules.

### PR DESCRIPTION
**Pull request by michaeljb**
Locally, I tested multiple variants with a company that was a copy of LA Title, and selecting one or the other or both all worked correctly.

Multiplayer:
Screenshot from 2020-10-18 17-28-25

Hotseat:
Screenshot from 2020-10-18 17-28-33

Game cards, one created with LA Title and one created without:
Screenshot from 2020-10-18 17-29-14

This PR replaces/is a follow-up to #1889. Fixes #1082. /cc @perwestling

### CODE REVIEW

**Comment by michaeljb** [comment]
@tobymao I just now saw your comment on #1889 about holding off on variants, should I close this PR down?

**Comment by tobymao** [comment]
let me look at the pr, again, if it's good i guess we can continue

**Comment by perwestling** [comment]
I think this interface looks better than the first attempt. I would like to see how it acts with 2 or 3 choices as well, where all is chosen. 
But I think I can try this PR on 18AL to see how it behaves for this.

File: assets/app/view/create_game.rb
Comment (tobymao) [inline assets/app/view/create_game.rb:14]: should we just call this optional_rules?

File: assets/app/view/create_game.rb
Comment (tobymao) [inline assets/app/view/create_game.rb:160]: should we have one method that sets all the stuff, like num players and optional rules?

File: models/game.rb
Comment (tobymao) [inline models/game.rb:107]: hm i wonder why we don't just always include the settings

**Comment by perwestling** [comment]
Have now tried to  apply the 4 optional rules of 18AL to the above branch (locally for me) and it seems to work fine. Selected all 4 and they work fine. 

I have added a printout method in 18AL that prints out the short name (description) of each optional rule used in the game. Not sure this should be used in general.

File: models/game.rb
Comment (michaeljb) [reply models/game.rb:107]: I'll probably have time to address comments tomorrow, want me to just make it always include settings?

**Comment by michaeljb** [comment]
> I have added a printout method in 18AL that prints out the short name (description) of each optional rule used in the game. Not sure this should be used in general.

@perwestling I think that sounds like it'd be worth adding for general use. If you want to push it up I could integrate it with this branch, or we could wait till this one is merged in :slightly_smiling_face:

File: models/game.rb
Comment (tobymao) [reply models/game.rb:107]: maybe

**Comment by perwestling** [comment]
I have pushed a branch to my fork: https://github.com/perwestling/18xx/pull/new/feature/optional_rules/add_18AL

The print code can be found in g_18_al.rb

File: assets/app/view/create_game.rb
Comment (michaeljb) [reply assets/app/view/create_game.rb]: `update_inputs` does this now

File: models/game.rb
Comment (michaeljb) [reply models/game.rb]: I did this, settings is now always included, but `seed` is only included if the actions are included

**Comment by michaeljb** [comment]
Link to all changes I've added since @perwestling's c00c92bf - https://github.com/tobymao/18xx/pull/1930/files/c00c92bfb3e3be9214a4338dbaf907880ef48393..eeecda97136d5cdb1fff3d8ddec51ce1ee1565e8

**Review by perwestling — COMMENTED** [review]
Is this part of the commit?

**Comment by perwestling** [comment]
It is a bit messy when it is so many commits....

I am not sure it was a good idea to add 18AL to this commit. Some of the general code changes that was needed (mainly accessors) was due to 18AL optional rules needs, and that might be clearer if having that in a separate commit. 
Oh well, this PR should be a good pilot for optional rules.

I will wait with looking into optional rules for 18MEX until this is approved.  They look pretty straightforward to implement so these we can add later.

**Comment by michaeljb** [comment]
With the [squash and merge](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#squash-and-merge-your-pull-request-commits) option you can always make a single commit on master when merging a PR, no matter how messy the branch :slightly_smiling_face: 

I've pushed up some commits reverting all of the 18AL pieces, so now this PR isn't adding any of 18AL's optional rules.

**Comment by perwestling** [comment]
Ok. I will apply the fixes to the 18AL optional rules commit when it is time to make a PR for that. At least we have tested that it works fine with multiple optional rules.

File: models/game.rb
Comment (tobymao) [inline models/game.rb:108]: why do we delete seed? just keep it in. once you make this change go ahead and merge

**Review by tobymao — APPROVED** [review]

File: models/game.rb
Comment (michaeljb) [reply models/game.rb:108]: That was the one thing I saw that was really being excluded by the old logic of only including settings when actions were also included. Will make this change when I get back to a computer...

**Pull request merged**
